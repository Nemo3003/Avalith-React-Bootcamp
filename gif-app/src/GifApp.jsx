import React, { useState } from "react";
import { AddCategory } from "./components/AddCategoty";
import { GifGrid } from "./components/GifGrid";
import { Navbar } from "./components/NavBar";
/*import './css/small.css';
import  './css/large.css';*/


export const GifApp = () => {
  const [categories, setCategories] = useState([]);

  const handleAddCategory = (value) => {
    setCategories([value, ...categories]);
  };
//This function is used to delete the category
  const handleRemotion = index => {
    const value = [...categories];
    value.splice(index, 1);
    setCategories(value);
  };

  
  return (
    <>
      {/* Titulo */}
      <h1 className="title-app">GifApp</h1>
    <Navbar/>
      {/* Input */}
      <AddCategory addCategory={handleAddCategory} />
      {/* Listado de Gif */}
      {
      React.Children.toArray(
      categories.map((category) => (
                <><button className="button-30" role="button" onClick={handleRemotion}>Delete Category</button><GifGrid className="categories" key={category} category={category}/></>
      ))
    )}
    </>
  );
};