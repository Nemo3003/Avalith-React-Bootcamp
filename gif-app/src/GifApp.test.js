import { GifApp } from "./GifApp";
import { renderHook, act } from "@testing-library/react";
//Verificar cambio en el estado de categorías, cuando se ejecuta la función handleAddCategory.
describe('Testing the main component GifApp', () => {

    test('Debe retornar el estado inicial', () => {
        const { result } = renderHook(() => GifApp());
        expect(result.current.categorias).toEqual([]);
    })

    test('Debe agregar una categoría', () => {
        const { result } = renderHook(() => GifApp());
        act(() => {
            result.current.handleAddCategory('One Punch');
        }
        )
        expect(result.current.categorias).toEqual(['One Punch']);
    })
});
    
