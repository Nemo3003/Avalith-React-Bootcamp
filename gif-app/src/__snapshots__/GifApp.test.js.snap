// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the main component GifApp El estado cambio 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search a Gif"
      type="text"
    />
  </form>,
  <button
    className="button-30"
    onClick={[Function]}
    role="button"
  >
    Reset
  </button>,
]
`;
